src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -418,6 +481,12 @@ public void onSuccess(List<RepairSessionResult> results)
             {
                 try
                 {
+                    if (results == null || results.stream().anyMatch(s -> s == null))
+                    {
+                        // something failed
+                        fail(null);
```

Comment:
"Something failed" is not very useful to the operator. Can we instead have a descriptive message so the operator can tell the difference between failing at this location vs failing at line [634](https://github.com/apache/cassandra/pull/446/files#diff-4bc513a60150419e20a9449c70a64a66R634)?

Reply:
In this case no without a larger rewrite. All paths of repair rely on list of futures to future as list where null means “error”. To propagate error is more invasive and would prefer in the JIRA about repair hanging.

I did change things a little bit. The errors are sent via notification (lossy) so may see a more detailed error; this is a fall back if notification was lost. 

I test with and without notification for reasons like this